tests/test_afwTable.py

```diff
@@ -0,0 +1,155 @@
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+
+import numpy
+import os
+import tempfile
+import shutil
+import sqlalchemy
+import unittest
+
+import lsst.utils.tests
+from lsst.daf.fmt.mysql import SqlStorage
+import lsst.daf.persistence as dafPersist
+
+
+def setup_module(module):
+    lsst.utils.tests.init()
+
+
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+
+class MyMapper(dafPersist.Mapper):
+
+    def __init__(self, root, *args, **kwargs):
+        self.storage = dafPersist.Storage.makeFromURI(root)
+        dafPersist.Mapper.__init__(root, *args, **kwargs)
+
+    def map_table(self, dataId, write):
+        loc = dafPersist.ButlerLocation(pythonType=lsst.afw.table.BaseCatalog,
+                                        cppType=None,
+                                        storageName=None,
+                                        locationList=['testname'],
+                                        dataId={},
+                                        mapper=self,
+                                        storage=self.storage)
+        return loc
+
+
+class TableIoTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.testDir = tempfile.mkdtemp(dir=ROOT, prefix="TableIoTestCase-")
+
+    def tearDown(self):
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir)
+
+    def test_create_false(self):
+        """Test that NoRepositoryAtRoot is raised if no repository exists and create is False"""
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        with self.assertRaises(dafPersist.NoRepositroyAtRoot):
+            SqlStorage(dbLocation, create=False)
+
+    def test_write_read(self):
+        """Test that Butler can write a BaseCatalog to an sqlite database, and that the database can be read
+        by sqlalchemy and compares equal to the original.
+        """
+        cat_expected = _make_afw_base_catalog()
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        butler = dafPersist.Butler(outputs={'cfgRoot': self.testDir, 'root': dbLocation, 'mapper': MyMapper})
+
+        # Test writing first
+        butler.put(cat_expected, 'table')
+        del butler
```

Comment:
Any need to close down db connection before closing down butler?

Reply:
The `del` is meant to close it (at the mercy of the garbage collector). We have recently talked about an explicit `close()` method (see DM-11801).